### PR TITLE
added Query ElastiCache with disabled at Rest Encryption

### LINKED ISSUE #820: Add ElastiCache with disabled rest encryption query for AWS Cloud Formation
### Platform
AWS Cloud Formation

### Provider
AWS

### Description
In order to protect sensitive data, AWS ElastiCache Redis clusters should be encrypted rest. Encryption of data at rest prevents unauthorized access to your sensitive data stored on AWS ElastiCache Redis clusters and associated cache storage.

### PR DESCRIPTION
**Pull request by cataraujo190**
Description:
In order to protect sensitive data, AWS ElastiCache Redis clusters should be encrypted rest. Encryption of data at rest prevents unauthorized access to your sensitive data stored on AWS ElastiCache Redis clusters and associated cache storage.

Close #820

### CODE REVIEW

File: assets/queries/cloudFormation/elasticache_with_disabled_at_rest_encryption/query.rego [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package Cx
+
+
+CxPolicy [ result  ] { 
+
+  document := input.document[i]
+  resource := document.Resources[key]
+
```
Comment (ruigomescx) [inline assets/queries/cloudFormation/elasticache_with_disabled_at_rest_encryption/query.rego:8]: At what types of resource does this query check? Type: 'AWS::Route53::RecordSet' and AWS::ElastiCache::ReplicationGroup? If so, we should check for the type to restrict the number of query results

File: assets/queries/cloudFormation/elasticache_with_disabled_at_rest_encryption/query.rego [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  document := input.document[i]
+  resource := document.Resources[key]
+  
+  properties := resource.Properties
+  lower(properties.Engine) == "redis"
+  properties.AtRestEncryptionEnabled == false 
+  result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    sprintf("Resources.%s.properties", [key]),
```
Comment (ruigomescx) [inline assets/queries/cloudFormation/elasticache_with_disabled_at_rest_encryption/query.rego:14]: can we refine our searchKey to point to the exact line? By checking the type of resource, we should be able to navigate to this field

File: assets/queries/cloudFormation/elasticache_with_disabled_at_rest_encryption/test/negative.yaml [hunk 3]
```diff
@@ -0,0 +1,452 @@
+AWSTemplateFormatVersion: '2010-09-09'
```
Comment (ruigomescx) [inline assets/queries/cloudFormation/elasticache_with_disabled_at_rest_encryption/test/negative.yaml:1]: The sample is too big, can we cut to only display the important parts?

File: assets/queries/cloudFormation/elasticache_with_disabled_at_rest_encryption/test/positive.yaml [hunk 4]
```diff
@@ -0,0 +1,594 @@
+AWSTemplateFormatVersion: '2010-09-09'
```
Comment (ruigomescx) [inline assets/queries/cloudFormation/elasticache_with_disabled_at_rest_encryption/test/positive.yaml:1]: The sample is too big, can we cut to only display the important parts?

**Review by ruigomescx — CHANGES_REQUESTED** [review]

**Review by joelsou5a — APPROVED** [review]

**Pull request merged**